spring-cql/src/test/java/org/springframework/cassandra/test/integration/EmbeddedCassandraServerHelper.java

```diff
@@ -0,0 +1,357 @@
+		InputStream is = EmbeddedCassandraServerHelper.class.getResourceAsStream(resource);
+		OutputStream out = new FileOutputStream(file);
+		byte buf[] = new byte[1024];
+		int len;
+		while ((len = is.read(buf)) > 0) {
+			out.write(buf, 0, len);
+		}
+		out.close();
+	}
+
+	/**
+	 * Creates a directory
+	 *
+	 * @param dir
+	 * @throws IOException
+	 */
+	private static void mkdir(String dir) throws IOException {
+		FileUtils.createDirectory(dir);
+	}
+
+	private static void cleanupAndLeaveDirs() throws IOException {
+		mkdirs();
+		cleanup();
+		mkdirs();
+		CommitLog commitLog = CommitLog.instance;
+		commitLog.getContext(); // wait for commit log allocator instantiation to avoid hanging on a race condition
+		commitLog.resetUnsafe(); // cleanup screws w/ CommitLog, this brings it back to safe state
+	}
+
+	private static void cleanup() throws IOException {
+		// clean up commitlog
+		String[] directoryNames = { DatabaseDescriptor.getCommitLogLocation(), };
+		for (String dirName : directoryNames) {
+			File dir = new File(dirName);
+			if (!dir.exists())
+				throw new RuntimeException("No such directory: " + dir.getAbsolutePath());
+			FileUtils.deleteRecursive(dir);
+		}
+
+		// clean up data directory which are stored as data directory/table/data
+		// files
+		for (String dirName : DatabaseDescriptor.getAllDataFileLocations()) {
+			File dir = new File(dirName);
+			if (!dir.exists())
+				throw new RuntimeException("No such directory: " + dir.getAbsolutePath());
+			FileUtils.deleteRecursive(dir);
+		}
+	}
+
+	public static void mkdirs() {
+		DatabaseDescriptor.createAllDirectories();
+
+	}
+
+	private static void readAndAdaptYaml(File cassandraConfig) throws IOException {
+		String yaml = readYamlFileToString(cassandraConfig);
+
+		// read the ports and replace them if zero. dump back the changed string, preserving comments (thus no snakeyaml)
+		Pattern portPattern = Pattern.compile("^([a-z_]+)_port:\\s*([0-9]+)\\s*$", Pattern.MULTILINE);
+		Matcher portMatcher = portPattern.matcher(yaml);
+		StringBuffer sb = new StringBuffer();
+		boolean replaced = false;
+		while (portMatcher.find()) {
+			String portName = portMatcher.group(1);
+			int portValue = Integer.parseInt(portMatcher.group(2));
+			String replacement;
+			if (portValue == 0) {
+				portValue = findUnusedLocalPort();
+				replacement = portName + "_port: " + portValue;
+				replaced = true;
+			} else {
+				replacement = portMatcher.group(0);
+			}
+			portMatcher.appendReplacement(sb, replacement);
+		}
+		portMatcher.appendTail(sb);
+
+		if (replaced) {
+			writeStringToYamlFile(cassandraConfig, sb.toString());
+		}
+	}
+
+	private static String readYamlFileToString(File yamlFile) throws IOException {
+		// using UnicodeReader to read the correct encoding according to BOM
+		UnicodeReader reader = new UnicodeReader(new FileInputStream(yamlFile));
+		try {
+			StringBuffer sb = new StringBuffer();
+			char[] cbuf = new char[1024];
+
+			int readden = reader.read(cbuf);
```

Comment:
`readden` is not a word.  Use `read` (which is pronounced like "reed" in the present tense or "red" in past tense) or perhaps, `numCharacters`.